### DISCUSSION TITLE
Breaking Intellisense

### DISCUSSION THREAD

**Discussion by kdlslyv** [community/community#9430] [Copilot Conversations]
Hello,

I'd love to keep using Copilot, as the suggestions are a great typing help for otherwise mundane typing tasks.
However, if it suggests something that is not correct - and that still happens extremely often (not a problem - it cannot read my mind I think) it's fairly annoying to discard the inline suggestion in VScode and have the Intellisense box pop-up like usual.

![image](https://user-images.githubusercontent.com/13890151/147756631-61ed07e8-36e2-41ea-ae15-df6dc5a0b4ff.png)

Using `Ctrl+.` does not bring it up either, it effectively blocks Intellisense from showing up at this point.

This is annoying enough to stop using Copilot entirely.

**Comment by ADataGMan** [comment]
I have the same experience in VSCode, where it will suggest a certain style for a CSS line and it's not what I want to use. The only way I've been able to work around this is by disabling copilot entirely which is also not what I want to do. I'd like to be able to dismiss it's help and get the regular intellisense suggestions back.

**Reply by Phasor** [reply]
im in the same boat. I just diabled copilot which I dont want to do but I need the intellisense.

**Reply by McCulloughRT** [reply]
Same here, Copilot in VSCode completely breaks intellisense suggestions, which due to typescript integration (in my case at least) are both powerful and borderline necessary. At this point Copilot is just actively making the dev experience worse because I need typed object property/method descriptions a lot more often than I need the (sometimes very good!) AI auto-completes.

**Reply by Luciden** [reply]
Same here.

**Comment by eddsaura** [comment]
Yeah, it fights a lot with Intellisense. And CSS intellisense is really powerfull. Idk how I would address that but we need an improve definitely. Anyway, thanks for the little friend, it usually works well with easy things.

**Comment by sonic1981** [comment]
Yeah it's a pain when this big leap forward isn't as good as the tools that already exist. copilot should have intellisense built into it, you'd imagine. Sometimes it's code suggestions are plain stupid. This problem has made me abandon using it, intellisense is more helpful than copilot so if I have to choose one, it's the standard intellisense.

**Reply by ghost** [reply]
+1 for not overriding all the intellisense that is already there (from other extensions ). 

Copilot now basically disables any coding tools and makes you rely just on copilot , and it's far from usable enough to be used like that .

**Reply by kdlslyv** [reply]
yup thats exactly what this feedback implies 👍

**Comment by kdlslyv** [comment]
lets make some noise here, the current copilot behavior is flawed and I'd argue nobody would say that this does not warrant a fix 😆   
I'm a bit worried that nobody working on copilot noticed this discussion yet 😭

**Reply by sonic1981** [reply]
Well the way to get this noticed is to get it upvoted

**Reply by eddsaura** [reply]
Yeah, let's push this somehow. I feel like it works a little but better now, maybe? Have you notice any improvements?

**Comment by beingminimal** [comment]
Hey everyone please join discussion here and vote also here. It's diff topic but requires attention to improve DX.
https://github.com/github/feedback/discussions/15533
Thanks

**Reply by eddsaura** [reply]
what we actually need is to make the topic as accurate as possible for them to move quickly.

**Reply by jorismak** [reply]
That's a complete different topic? Please don't go across random topics to try to get more attention to yours.

**Comment by zkenda** [comment]
It would be great if there would be an option to disable default behaviour and trigger copilot with the command.

**Reply by Hypnocrit** [reply]
If one uses the control space intellisense hotkey it should choose intellisense.  Super annoying.

**Comment by gotmike** [comment]
perhaps this should be a separate post, but i also notice that copilot will sometimes recommend "almost" correct code, but it messes up the capitalization, so the code isn't correct. intellisense gives me the correct code with correct capitalization, but copilot overwrites it. so i think in many cases, where i see copilot fighting, it's really over capitalization. and it fights not just on my customized classes and methods, but standard microsoft ones too.

like intellisense would suggest `SqlUpdate.GetEmptyDataSet();` but copilot would suggest `sqlupdate.getemptydataset()` -- which is strange.

**Comment by jfaz13** [comment]
Adding interest to this. Now that Copilot costs money I've decided to just use Intellisense since the lack of this feature already had me on the fence. There should be a way of prioritizing what shows. E.G: Prioritize Copilot, and press a hotkey to bring up standard Intellisense when you don't like the suggestion, or prioritize Intellisense, and press a hotkey to bring up Copilot. I don't know the inner workings of Copilot, but if there's a confidence value then that could also be used to determine whether to bring up Intellisense/Copilot. Whatever the option, as it stands right now it's completely broken.

**Comment by jorismak** [comment]
yeah, when I got the notification 'Copilot is not setup yet' I was confused. Then I saw the 'start your trial' and the lightbulb went off in my head 'oh, is the experimental phase over? But they didn't fix intellisense yet? Nope, will not be using it then' and didn't look back.

The demo's seem cool, but in day to day coding live it hasn't helped me once and was annoying enough to finally turn off... so something's not quite there yet.

Just fixing the 'annoying' part, means it's OK to leave on. And then that day might come when it suddenly suggests something that makes me go 'oh wow'. But since it's now in the annoying-category, you turn it off, and you never learn what it might do for you.

Feels stupid that people with IntelliJ IDE's get a better product compared to 'their own' editor in VSCode.

**Comment by AlleywayDave** [comment]
Why is this still not fixed? Have the Copilot creators not tried using Copilot before? Asking for a friend.

**Reply by endasil** [reply]
Sadly copilot is not useful as it is right now as it keeps killing the IntelliSense suggestions and then just inserting a new row when I press return instead of the IntelliSense thing.

**Comment by Aaron-G-42** [comment]
I'm not sure why, but copilot in VS 2022 seems to interfere less with intellisense... seems like if you dismiss copilot suggestion in VS then you get to use intellisense. I just started using copilot with VSCode and it's unusable... intellisense must be accessible!

So, disabling copilot for VSCode until this is fixed.

**Reply by internetfx** [reply]
Unless you can prove a workaround for all the Visual Studio folks, don't post claims that it interferes less. I agree IntelliSense must be accessible, as everyone here agrees. Please, post a video demonstrating that Visual Studio provides C++ inline suggestions when Copilot is dismissed, and then do that with every other language. Back up your claim.

**Reply by AlleywayDave** [reply]
I rarely had problems with IntelliSense until I started using Copilot. I can occasionally see, for example, the dropdown for IntelliSense be shown at the same time as the Copilot suggestion. After a couple hours of coding, IntelliSense breaks and I only get the Copilot suggestions. Sometimes deleting the .vs folder fixes it, sometimes it doesn't. I didn't have these problems before.

**Reply by jorismak** [reply]
> Unless you can prove a workaround for all the Visual Studio folks, don't post claims that it interferes less. I agree IntelliSense must be accessible, as everyone here agrees. Please, post a video demonstrating that Visual Studio provides C++ inline suggestions when Copilot is dismissed, and then do that with every other language. Back up your claim.

No need to go on the offensive. It's another '+1' post, nothing more, nothing less. 
@Aaron-G-42 states that the Experience in Visual Studio is quite doable, but then was surprised that Visual Studio Code (completely different product and the subject of this issue) indeed has a big issue here.

That being said, having to dismiss copilot to access the intellisense that I'm expecting seems like a slowdown as well.

**Reply by internetfx** [reply]
Look at the posts. IntelliSense is in both products and the interference with IntelliSense is the subject of this issue whether or not the interface, behavior or functionality is dissimilar, so consider the topic expanded: the behavior described is what Visual Studio does also and it needs to be fixed. No offense intended. I'm sure we'll see a fix in an update if the Copilot team will address it.

**Comment by Harry-Loud** [comment]


**Reply by jorismak** [reply]
> So as such, this is not an issue with Copilot, but instead with VSCode itself.

Hmm... how is that?
If it's the Copilot extension that clears and uses the VSCode Intellisense listing, then I see that as problem in Copilot?

AFAIK, there are other extensions that manage to _add_ Intellisense suggestions without clearing / overruling the normal list.

If it's a VSCode limitation that only a single extension can provide Intellisense - and yes that means I'm wrong 😉  - then I see where the problem is coming from, and you could see that as a VSCode limitation.

**Reply by Harry-Loud** [reply]
> 

What I meant by that was that you cant have inline and Intelisense at the same time. Just one or the other.

**Reply by Harry-Loud** [reply]
All Copilot needs to do is not clear the IntelliSense list

**Reply by Harry-Loud** [reply]
If you want to toggle inline sugestions on and off, you can use [SettingCycler](https://marketplace.visualstudio.com/items?itemName=hoovercj.vscode-settings-cycler)
And this keyboard binding in Keyboard.json
`{
    "key": "ctrl+oem_5",
	  "command": "settings.cycle",
	  "args": {
		"id": "toggleInlineCopilot", // must be unique
        "overrideWorkspaceSettings": true,
		"values": [ // Note: use the same settings in each values object
			{
				"github.copilot.inlineSuggest.enable": true
			},
			{
				"github.copilot.inlineSuggest.enable": false
			}
		]
	}  }`

**Comment by Harry-Loud** [comment]
After more research, VSCode has built-in support for exactly this. [Inline completions with IntelliSense at the same time](https://github.com/microsoft/vscode/issues/129495).
Preview Image from this thread: ![](https://user-images.githubusercontent.com/2931520/129227447-7d77b9bd-03b8-4c0a-b00e-2a6114455f5e.gif)
It just needs to be integrated properly with Copilot.
This would so add to the functionality of Copilot.

For the time being. Ctrl + Space brings up IntelliSense when required.

**Reply by jorismak** [reply]
It's still _one_ inline completion, and then multiple Intellisense options.

Copilot also has more options (and the first is often not that useful), so in what you are saying it could only list one (the inline one), and then also multiline-suggestions can't be properly displayed?

In my last experience, Copilot in VSCode is (was?) unusable because of this. It's just not up to par (or too slow) to replace the regular intellisense, but it does completely hide it.

**Reply by jorismak** [reply]
Now that I think about it, it might work better (for me at least) so leave all the intellisense / auto-suggestion alone, and add a shortcut key to open the Copilot suggestion list. It could then even list the inline suggestion while I type.

At the moment, my typing is 'halted' and slowed down because the suggestion list doesn't pop-up when I expect it to (simple things, like completing function names, imports and accessing object properties/members). This makes my coding super slow instead of speeding it up, and the major reason for not using copilot anymore.

It has during testing maybe once., or maybe twice gave a suggestion where I thought 'yes, that is in line of what I want, I'll edit that, thank you'. Most of the time it's way off. And if it then hides the default suggestions you're used to using, it's a disaster.

So, let it give an inline suggestion _and_ open up more copilot suggestions on a special key press. That way it doesn't interfere with the normal behavior, and it can actually _add_ to the experience instead of subtracting things and then trying to replace them.

Pressing ctrl+space to open the suggestion list is not an option.

1) for me, it would be ctrl+i because ctrl+space is used by the OS 
2) I use the suggestion list multiple times per line that I write. Having to constantly press a shortcut to open it is also not speeding things up.

**Reply by Harry-Loud** [reply]
Its interesting to think how it could all be integrated.
I think you slightly missunderstand my origional post.
What I was trying to get at is VSCode has the capability to open up regular, normal IntelliSense, and when you hover over a sugestion, Copilot offers an Inline Sugestion prefixing the IntelliSense suggestion. When you click tab once, it accepts your InteliSense sugestion as normal, and twice to also get the Inline Copilot suggestion after it. This could be done right now with the existing VScode, the Copilot team just have to integrate it.

Even better and more separated I think would be to have Copilot show via Inline, and IntelliSense via its normal box at the same time, without interference. Then one could use a different keystroke (tab vs enter maybe) to accept whether Copilots Inline Suggestion or the Regular IntelliSense suggestion.

When It comes to MultiLine Copilot suggestions, maybe that could be restricted to when InteliSense has no suggestions (at the end of a function bracket like this: function DoThing(var1: boolean) { |<-Cursor Here  ). Because thats when Multiline suggestions usually work best. And failing that, even just restrict it to Ctrl + Enter.

There are a few different ways of doing it. Either would work, because as is, Copilot is completely broken and interferes with every programmer's flow more than it helps.

On a personal note, I have installed the "Settings Cycler" extension and cycle the Copilot Inline suggestions on or off with Alt + \\.

**Comment by indoflaven** [comment]
Had to disable Copilot in VS 2022.  Kept getting the same Copilot suggestion, kept hitting ESC and could never get Intellisense to appear!

**Comment by mcgrue** [comment]
Same as all the above.  I really want to see what copilot suggests in it's own window and have a hotkey to apply it WITHOUT clobbering intellisense.

Until it has that functionality I have to disable it and will not be buying it.

**Reply by mcgrue** [reply]
I believe if you had a configurable "wait this many milliseconds of no-input before making an inline suggestion" that would ALSO solve the issue, and quite quickly.  Since you already have a hardcoded value for the "wait for pauses in typing before showing suggestions" this SHOULD be an easy add by making that a userconfig variable.

![image](https://user-images.githubusercontent.com/213483/192059121-a65984f5-3695-4223-8610-3fcc48b1a6d7.png)

**Comment by AyoParadis** [comment]
is there any work being done on this for VS Code?

**Comment by Gumball12** [comment]
_any update?_

**Comment by is-it-ayush** [comment]
I think it'd be great to trigger co-pilot via a keyboard bind instead of it auto-suggesting. Sine both intellisense and co-pilot are supposed to the same thing (i.e. suggest) they end up fighting. It'd be great to setup a keyboard bind to trigger co-pilot suggestions when needed while prioritizing intellisense. The other solution could be to somehow train intellisense inside co-pilot for most major languages out there. : )

**Reply by Harry-Loud** [reply]
Alt + Space does something similar for me

**Comment by CoryDHall** [comment]
Adding commentary/suggestion:

Having worked across 5 discrete document language types in the past week with copilot, the core issue is that the AI is actually "code dumb". 

Like a student that is a good test taker, but has no actual knowledge of the subject.

The proposition is that based off of a combination of filetype, partial file contents, and the characters most recently typed into the editor buffer, it can infer enough intent to stub out appropriate boilerplate—in theory faster than waiting for the language server to report the appropriate AST symbols to the intellisense result filtering algorithm.

The problem(s) is the language servers, which are not just aware of user input, but also workspace/project structure and declared APIs, so they can stop narrowing when a reasonable and somewhat validated suggestion list is available—because sometimes, there _is_ more than one possible _right_ answer. 

The copilot AI's loss gradient only considers whether you erased their suggestion, with no consideration for current AST Node information or Intellisense suggestions. If it did, then its core proposition would be to write the code you _intend_ to write vs the code you are _likely_ to write.

I _believe_ the copilot AI is simply incapable of understanding anything more than editor buffers (it is after all, trained on static public repository files), *but* I think (at least for me) it would be helpful if the VSCode extension allowed a black/white-list based on the [current AST symbol](https://code.visualstudio.com/docs/editor/intellisense#_types-of-completions). My most common use case is Copilot being unaware of new files in the directory since those are created outside of the editor buffers. I *_NEVER_* want copilot suggestions while I'm typing a _path_ (`file`, `directory`, `module`) because I already know and the editor _inherently knows_ where the file is, I don't need a guesstimate, I just want the convenience to essentially fuzzy type the path. 

And this is a common pattern with all of my usability issues with Copilot—there are some things that a developer absolutely knows ahead of time, and uses intellisense not to find the solution, but to reduce keystrokes and streamline _writing_ the solution. I also believe this hubristic behavior is to Copilot's detriment because it's missing the opportunity to ingest valuable/expensive context metadata that is freely available.

**Comment by ivanTemelkov** [comment]
I also have noticed that Copilot and Intellisense doesn't work well in Visual Studio 2022 with ReSharper.

Another annoying issue is that when I start typing, for example, "builder.serv" (lowercase s) the Intellisense kicks in suggesting "Services" (uppercase S) I hit TAB and waht I get is builder.services (lowercase s) that is wrong.

![image](https://user-images.githubusercontent.com/43280639/199437651-96b180a2-7a22-4b0c-8999-5d740ad84246.png)

Looking at the image it is obvious that even if the suggestion to use Syncfusion could be correct, the syntax (using lowercase s for Services) is incorrect.

I don't think that at this stage Copilot is of any help since it rather breaks the productivity than enhance it.

**Reply by gotmike** [reply]
I can second this issue. The non-capitalization thing is quite annoying. I didn't actually realize it was a conflict with ReSharper. If you uninstall ReSharper does it properly capitalize?

**Reply by ivanTemelkov** [reply]
Hi,
I'm sorry, I cannot help you with this.
I already suspended my Copilot trial and uninstalled the extension from VS 2022.
I find it very disruptive.

Best regards,
Ivan

**Comment by Aitchy13** [comment]
I've found this quite frustrating too. The only "workaround" I have in situations where you're accessing properties of an object is to press "Esc" when you see the CoPilot suggestion before you press "." to access the list of available properties.

**Comment by AyoParadis** [comment]
James’s solution is probably the best work around we’ve got.
On Thu, 24 Nov 2022 at 20:29, James Aitchison ***@***.***>
wrote:

> I've found this quite frustrating too. The only "workaround" I have in
> situations where you're accessing properties of an object is to press "Esc"
> when you see the CoPilot suggestion before you press "." to access the list
> of availability properties.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/community/discussions/9430#discussioncomment-4230434>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARF2KVIQPW6W42LDMTVBLJTWJ66YVANCNFSM5K7XOGMQ>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>
-- 
Kind Regards,
Ayo Adesanya
https://www.ayoadesanya.com/

**Reply by ak5** [reply]
This doesn't work with vim bindings

**Comment by Pckool** [comment]
Okay so it seems that I can get intellisense to show up (on my configuration) by pressing `CTRL+SPACE` when I see the copilot prompt/suggestion. This works pretty well for me

**Reply by jonahkagan** [reply]
This works for me too! Great find

**Reply by cvcore** [reply]
For me `ctrl+space` didnt't work on mac but `cmd + I` does the job. One needs to check what's the keybinding for action `toggleSuggestionDetails` on the system.

**Comment by jorismak** [comment]
This ruins my flow completely , to manually have to activate something that
I'm used to popup automatically.

Specially since i have to use Ctrl+i , because Ctrl+ space is a system
shortcut.

**Comment by ShackMan2000** [comment]
Still the same in Visual Studio. Having to press ctrl + space all the time to get rid of the wrong copilot suggestions and see IntelliSense kinda destroys all the speedboost I get from Copilot.

**Comment by joenot443** [comment]
Considering this is presently the highest upvoted unanswered discussion item in the queue, it'd be great to get some communication from the devs on this topic. Now that I'm back to working in an environment where Intellisense is pretty heavily relied upon, Copilot's become effectively unusable. I'm planning on unsubscribing next month largely due to this, which is unfortunate, considering how great the tool is otherwise.

**Reply by whenmoon** [reply]
Same.

**Comment by mcgrue** [comment]
Same.  I've been monitoring this thread and... I'm about to unsubscribe due
to this (very easily solved with a configurable timeout setting) issue.

**Comment by ghost** [comment]
Could copilot not check if intellisense would suggest something, and if so, it could remove its own suggestion? 
Or does VSCode not allow such interaction?

**Comment by jorismak** [comment]
In the 'Whats New' for VSCode 1.75 is a special mention of Copilot, and
that they now make use of a (new?) inline suggestion UI.

[image: image.png]

Did anyone try it out if it's an improvement? Because then it might be
worthwhile to try Copilot again.

**Comment by ohabash** [comment]
`ctrl+space` is a great work around. However still for more harmony between CoPilot and intellisense in the future.

**Reply by webbugt** [reply]
I'd love to be able to configure inverse wher intellisense has priority and I have to use shift+enter to get copilot's suggestion

**Comment by JamesTrotter-TBADonc** [comment]
Have had to disable co-pilot. Far too intrusive. Probably will unsubscribe from the service as Visual Studio is my main use case.

**Comment by webbugt** [comment]
Shouldn't this be an easy fix?
- Defer to intellisense if it has suggestions
- wait for shift+Enter (or delay. Make it configurable) to start copilot suggestions

Yeah it would be nice if Copilot took intellisense suggestions when generating. Maybe it does. But as it stands. It's intruding on a more stable and precise tool we already have.

Another idea is that if developer rejects a suggestion, show intellisense suggestion next

**Comment by whenmoon** [comment]
A year and three months later this is still happening. But now it seems to override TypeScript autocompletions too. It's virtually unbearable.

**Comment by WMcKibbin** [comment]
You can hit escape to drop the copilot suggestions which I definitely want as the first thing because I end up using it so frequently. If I drop the suggestion, show me the intellisense please.

**Comment by NavidMitchell** [comment]
I think it would nice if the copilot suggestions appeared below the line I am typing and the regular autocomplete appeared on the same line. Then give me a key to accept either depending I which I need.

**Comment by RT-Tap** [comment]
I'm having the same issue.  
However I think the real issue here is copilot is integrating itself with inline suggest capability of vscode.  It should be more like an addition rather than an integration if that makes sense.  For example I want to have the keybinding `tab` to accept intellisnse/vscode built in inline suggest  and then `shift+tab` for copilot completion however I cant even do that because copilot just forces itself into inlinesuggest,  you need to change the  `editor.action.inlineSuggest.commit` keybinding to change copilots accept.  Why is this the case these need to be separated and treated as separate things.

I have a tabout extension which takes me out of a closing brace/parenthesis by pressing tab , copilot just ruins this for me as well. 

Maybe like leave inline commit/intellisense alone and then have copilots suggestions as a different shade/color so you can distinguish which is which and choose from either option.

The reason I suggest separating out copilot from inline suggestions is because ultimately it's an aid were not using it to write every single line of code so just why the replacement/integration with an already good/working tool (intellisence)

**Reply by endasil** [reply]
A year has passed and this bug is still here. Is copilot going to be useless forever?

**Comment by eddsaura** [comment]
Is this post still being relevant? Has the team reached to this?

**Reply by HeatherFlux** [reply]
Just came here to ask the same thing. I have to turn it off for now until the fix it. At least giving us a black list of characters that we don't want copilot to respond to would be nice. Or integrate it into the intellisense.

**Comment by ProTip** [comment]
I believe I'm going to need to disable and cancel my trial based on working in VSCode with C# and TypeScript. Sometimes the scaffolding suggestions are great but I find the single line ghost text to be wrong 75%+ of the time.

It's maddening having to fight with it for autocomplete; hitting `control+space` only to have autocomplete disappear again if I backspace or change the line :( Wish I could trigger copilot on demand or show both at the same time.

I realize people work differently but it's.. Interesting that they seem to be doubling down that they've got the correct behavior based on data from their user trials haha. Sampling bias?

**Reply by ggrantrowberry** [reply]
I found this today. This seems to be a good solution until they fix the problem. [Stackoverflow](https://stackoverflow.com/questions/71224911/can-github-copilot-stop-auto-suggesting-instead-be-triggered-by-a-keystroke)

**Comment by HeatherFlux** [comment]
Will this be fixed? Can I just have to autocomplete suggestion show up in the intellisense?

**Comment by GitMurf** [comment]
Any updates or even acknowledgment by anyone on the copilot team? Quite frustrating the UX/DX has not been improved on this since day 1 :( I LOVE copilot but man is this frustrating!

**Comment by jorismak** [comment]
I now only use copilot chat. Disabled inline suggestions for copilot.

**Comment by XeiDaMoKa** [comment]
in here for the opposite , i can't get copilot to suggest because of the css tool tip suggesting

**Reply by cutecycle** [reply]
Same, actually! like sometimes I want copilot to suggest but Intellisense butts it out.

**Comment by isidorn** [comment]
Hi, VS Code PM here 👋  @hediet has fixed this issue 🎉   
Can you all please try it with latest VS Code and latest Copilot and let us know if there are still issues that you are hitting. If yes, please describe precisely what seems to be the problem.

Thanks all 🙏

**Reply by jorismak** [reply]
wow, didn't expect a reply after all this time (over a year). But I'll for sure enable it again and test some more.

**Comment by GitMurf** [comment]
> Hi, VS Code PM here 👋 @hediet has fixed this issue 🎉 Can you all please try it with latest VS Code and latest Copilot and let us know if there are still issues that you are hitting. If yes, please describe precisely what seems to be the problem.
> 
> Thanks all 🙏

@isidorn Thank you for the update but can you provide some clarity into how it was fixed and/or what exactly the solution was? So we know what to expect and test. Is there another gh issue item and/or PR item we can see/reference for details?

**Reply by isidorn** [reply]
https://github.com/microsoft/vscode/issues/185831
https://github.com/microsoft/vscode/issues/131940

**Reply by Manouchehri** [reply]
I'm still kinda confused, it doesn't seem to be fixed from what I can tell..? What version(s) should be fixed?

**Reply by isidorn** [reply]
Latest VS Code stable and latest Copilot. Can you please describe precisely what seems still to be the problem? Thanks

**Comment by jorismak** [comment]
Well, I don't get copilot anymore in the intellisense popup it seems. I
only get an online suggestion. Which needs some getting used to.

So at least intellisense isn't broken anymore , And copilot can stay
enabled.

I'm still used to using it on demand in the copilot chat window and not
inline in the editor. But I'll leave it on to see if it comes up with some
things.
(I've only seen very wrong inline suggests at the moment :)).

**Comment by bawantha** [comment]
This  co-pilot thing suppose to help developers and now this is.

**Comment by jorismak** [comment]
The original problem was that CoPilot inserted itself in the Intellisense popup, and thereby removes any other Intellisense providers that you were used to. You **_only_** got Copilot things in Intellisense.

What now seems to have happened, is that there are no Copilot suggestions at all anymore in Intellisense. I _only_ get the Copilot 'inline suggestion', that grey text you get inside your editor.

This is nice, because it doesn't removes the other simpler Intellisense that you still want.
But it's not nice, in that you only get the single Copilot suggestion, which is - in my experience so far - never ever correct.

I guess I was hoping that Copilot suggestions would popup in Intellisense, sorted at the bottom. That if you take the time to stop and look at it you can choose it, but if not you can just continue with Intellisense and code like you normally have.

So, I'm happy Copilot isn't messing up the existing Intellisense anymore (it's been over a year!) but it has made Copilot less capable it seems :(.

**Reply by GitMurf** [reply]
wooo wait a second, you got copilot suggestions in intellisense in the past?!! I have been using it since day one of beta and I have always only gotten the inline suggestion! My issue was that the inline suggestion would override what you were trying to do in intellisense so if you hit tab for intellisense item it would override it and just give you the copilot inline suggestion. That is what I thought this entire issue/bug was about here... and that has now been fixed it appears.

BUT I want copilot suggestions in my intellisense haha! How can we get that? I never knew that was even a feature.

**Reply by isidorn** [reply]
That is not a feature and was never the case. @jorismak is incorrect.

**Reply by GitMurf** [reply]
ok @isidorn thank you for clarifying! haha. So then is the rest of my description accurate on what the assumed problem was in the past? and what has recently been improved? Essentially where the suggestions no longer override intellisense and will only be applied if you close out of intellisense?

**Comment by jorismak** [comment]
Ha, lol. Sorry. but the issue is called "breaking Intellisense", because
Copilot broke the regular Intellisense, because it injected stuff there and
removed what was there without Copilot .

So no, I'm not incorrect. Maybe that has been fixed way earlier (I have
Copilot disabled since then because there was no official response that
this was going to be fixed). There was even discussion about the fact that
the IntelliJ plugin opened _two_ suggestion boxes. The regular
Intellisense, and the Copilot next to it (and why that wasn't possible with
vscode).

And maybe the discussion in this topic that is quickly becoming _two_ years
old has drifted to other problems, but the original problem was exactly
that: Regular Intellisense was _replaced_ with Copilot Intellisense, and
that killed the workflow for a lot of people.

In the current implementation, I see that Copilot only gives an inline
suggestion, which is better. The inline suggestion is selected with tab,
Intellisense is (and always has been!?) selected with 'Enter' so this seems
fine. I just miss the other suggestions Copilot used to give.
You can still open them with ctrl+enter ('open completions panel' command)
but short one-lliners were in the Intellisense before (I'm talking about
the first pre-release invite-only period).

You could also write a comment with an instruction, like '// write a method
that fetches all objects of type ' and it would give an inline multi-line
suggestion of that. Now it just wants to write a single line, and often
it's just completing the comment instead of generating the method.

You _can_ now open the Copilot Chat function, to type '/generate a method
that fetches all objects of type' and that will actually be way more
correct most of the time. But before it used to do this in the inline
suggests. Maybe that's still possible with settings in the
github.copilot.advanced section, I don't know.

**Comment by asheroto** [comment]
This is still an issue. Occurs in all areas. This is a simple example. My WriteDebug function accepts one string as a parameter. Uninstalled Visual Studio 2022 completely. Reinstalled completely. Updated all extensions.

Note around second 27.

https://github.com/community/community/assets/49938263/86bbe08a-b825-4bde-9459-403fc4afe414

**Reply by wwmike** [reply]
Yeah, I stopped using copilot for this reason.

**Comment by evgeniiarts** [comment]
January 2025 and similar issues still persist.
Having a small typescript project and enabling Copilot as the only extension results in "Loading IntelliSense status..." forever (well, it manages to finish after 10 minutes sometimes but as soon as you make some changes... it starts over).

Win11, newest updates.
32GB hardware RAM and additional 64gb virtual RAM added. 
I have also followed the suggestion here:
https://stackoverflow.com/questions/66852458/how-can-i-increase-cpu-ram-available-to-vscode#:~:text=You%20can%20increase%2Fdecrease%20the,for%20your%20desired%20maximum%20RAM.
and increased VS Code's Max TS Server Memory to 4028.

The only fix I found was to uninstall Copilot despite having purchased a yearly license recently. Quite sad but better then dealing with it after my experience so far.

**Reply by jorismak** [reply]
These days I dont notice issues anymore, copilot as intellisense suggestions seems to be completely removed? (Or I turn it off without knowing). 

If intellisense doesn't load (typescript support is builtin in vscode) it's probably something wrong with tsconfig.json.

My typescript projects are often scaffolded by Nuxt or Wrangler (Cloudflare) and never had issues .

Doesn't mean your issues aren't real of course , but I'd look into checking other stuff and not blaming copilot . Not because you are not allowed, but because I honestly think it stems from somewhere else.

**Reply by evgeniiarts** [reply]
@jorismak Thanks for you reply. It made me double check my setup once again - and I have reinstalled VS code, and have installed copilot as the only extension. 
To my surprise, because I had disabled all extensions before and restarted VS Code, it worked this time.
The IDE is as fast as expected and works with copilot. I kept my setup like that and did not install any other extensions.

But... after an hour or so the "Loading IntelliSense status...." issue came up again. And takes forever.
In addition, my VS Code provided this warning notification:
"The JS/TS language service crashed 5 times in the last 5 Minutes. This may be caused by a plugin contributed by one of these extensions: GitHub.copilot-chat Please try disabling these extensions before filing an issue against VS Code."

**Comment by DereC4** [comment]
Confirmed this still happens for my Typescript files

**Comment by sanengineer** [comment]
<img width="399" height="156" alt="Screenshot 2025-08-21 at 17 59 50" src="https://github.com/user-attachments/assets/21db583d-0c50-42b2-99d2-cbc203c601d8" />

intellisense not working when github copilot activated!

**Comment by StefanoVe** [comment]
I have the same issue, IntelliSense gets stuck loading whenever i ask something to copilot chat. The only workaround i've found is to reload the window after each answer

**Comment by DereC4** [comment]
[in the meantime check out this certified helldivers 2 meme i made](https://www.youtube.com/watch?v=POgWLYi21Wk)

**Comment by jdublon-correla** [comment]
Found a workaround - install the "Path Intellisense" extension by Christian Kohler. Only for paths obviously but that is the main thing I was missing from intellisense!

**Comment by jschank** [comment]
I have a similar problem, I don't know if it is copilot related or not.
When I'm editing code (in this case golang) I get what looks like an intellisense pop-up, but esc does not make it do away, in fact, NOTHING makes it go away, and when it doesn't have a choice I want, then the only way to dismiss it, is to accept something that is wrong, and then edit it away.

For example:
I am trying to declare a constant. I type cons, and the popup gives all sorts of choices, none of which are "const", and escape will not dismiss it. (rather it looks like it dismisses, but then opens instantly again)

This is completely unacceptable (*initiate rant about microsoft*) nor is it clear as to how to fix it.
